File: sources/im/getsocial/sdk/invites/LinkParams.java
package im.getsocial.sdk.invites;

import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Map.Entry;

public class LinkParams extends HashMap<String, Object> {
    public static final String KEY_CUSTOM_DESCRIPTION = "$description";
    public static final String KEY_CUSTOM_IMAGE = "$image";
    public static final String KEY_CUSTOM_TITLE = "$title";
    public static final String KEY_CUSTOM_YOUTUBE_VIDEO = "$youtube_video";
    /* renamed from: a */
    private static final List<String> f2282a = Arrays.asList(new String[]{KEY_CUSTOM_TITLE, KEY_CUSTOM_DESCRIPTION, KEY_CUSTOM_IMAGE, KEY_CUSTOM_YOUTUBE_VIDEO});

    public LinkParams(Map<String, Object> map) {
        super(map);
    }

    public static String validateLinkParams(LinkParams linkParams) {
        for (String str : linkParams.keySet()) {
            if (str.charAt(0) == '$' && !f2282a.contains(str)) {
                return str;
            }
        }
        return null;
    }

    public Map<String, String> getStringValues() {
        Map<String, String> hashMap = new HashMap();
        for (Entry entry : entrySet()) {
            Object value = entry.getValue();
            if (value instanceof String) {
                hashMap.put(entry.getKey(), (String) value);
            }
        }
        return hashMap;
    }
}

File: assets/bin/Data/Managed/Assembly-CSharp/GetSocialSdk.Core/LinkParams.cs
using System.Collections.Generic;
using UnityEngine;

namespace GetSocialSdk.Core
{
	public sealed class LinkParams : Dictionary<string, object>, IConvertableFromNative<LinkParams>, IConvertableToNative
	{
		public const string KeyCustomTitle = "$title";

		public const string KeyCustomDescription = "$description";

		public const string KeyCustomImage = "$image";

		public const string KeyCustomYouTubeVideo = "$youtube_video";

		public LinkParams()
		{
		}

		public LinkParams(Dictionary<string, string> data)
		{
			if (data != null)
			{
				foreach (KeyValuePair<string, string> datum in data)
				{
					this[datum.Key] = datum.Value;
				}
			}
		}

		public LinkParams(Dictionary<string, object> data)
		{
			if (data != null)
			{
				foreach (KeyValuePair<string, object> datum in data)
				{
					this[datum.Key] = datum.Value;
				}
			}
		}

		public override string ToString()
		{
			return $"[LinkParams: {this.ToDebugString()}]";
		}

		private bool Equals(LinkParams other)
		{
			return this.DictionaryEquals(other);
		}

		public override bool Equals(object obj)
		{
			if (object.ReferenceEquals(null, obj))
			{
				return false;
			}
			if (object.ReferenceEquals(this, obj))
			{
				return true;
			}
			return obj is LinkParams && Equals((LinkParams)obj);
		}

		public override int GetHashCode()
		{
			return base.GetHashCode();
		}

		public AndroidJavaObject ToAjo()
		{
			//IL_005b: Unknown result type (might be due to invalid IL or missing references)
			//IL_0060: Expected O, but got Unknown
			Dictionary<string, object> dictionary = new Dictionary<string, object>(this);
			if (dictionary.ContainsKey("$image"))
			{
				object obj = dictionary["$image"];
				Texture2D val = obj as Texture2D;
				if (val != null)
				{
					dictionary["$image"] = val.ToAjoBitmap();
				}
			}
			return new AndroidJavaObject("im.getsocial.sdk.invites.LinkParams", new object[1]
			{
				dictionary.ToJavaHashMap()
			});
		}

		public LinkParams ParseFromAJO(AndroidJavaObject ajo)
		{
			return new LinkParams(ajo.FromJavaHashMap());
		}
	}
}
